cmd/agent/windows/service/service.go

```diff
@@ -45,7 +45,7 @@ func (m *agentWindowsService) Execute(args []string, r <-chan svc.ChangeRequest,
 		return
 	}
 	elog.Info(0x40000003, config.ServiceName)
-
+	changes <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
```

Comment:
Do you know how long can the agent service can stay in StartPending state before SCM reports an error? (and what the effects of such an error would be?)

Reply:
it's not clear to me how long it will wait after the initial startpending.
You can/should extend the time by sending multiple StartPending with "updates" as to how long the scm should wait.
We should also dig into the go wrapper and see what, if anything, they're assuming for the "wait hint"

Reply:
That's helpful thanks. Were you able to reproduce a scenario where `StartAgent` is very slow, just to check the behavior of SCM?

(if not, I think this is something that should be QA'ed/tested. We could introduce a long `sleep` in `StartAgent` to see what the effects are)

Reply:
yes, it's quite simple.
In powershell just do
`restart-service -force datadogagent; restart-service -force datadogagent`.
The 2nd one fails, and you're left with a running trace agent but not core aent.